kahuna/public/js/search/query.js

```diff
@@ -62,6 +69,16 @@ query.controller('SearchQueryCtrl',
     Object.keys($stateParams)
           .forEach(setAndWatchParam);
 
+    ctrl.changeQuery = () =>
+        searchQueryService.set(ctrl.filter.query);
+
+    // FIXME: This is here to stop circular injection of the model.
+    // Avoiding: queryChange => set() => emit() => queryChange()
+    searchQueryService.q$.filter(q =>
+        q !== ctrl.filter.query
+    ).subscribe(q => ctrl.filter.query = q);
```

Comment:
[I'm using the inject now](https://github.com/guardian/grid/commit/dd56ac3da86f44699e549aeb97be6bda3b397792#diff-32fea343d36b21c11521e08ea0059cafR78), so it will be.